src/views/sources/addSourceModal.tsx

```diff
@@ -70,188 +105,220 @@ const AddSourceModal: React.FC<AddSourceModalProps> = ({
+            />
+            <HelperText>Enter an IP address or hostname</HelperText>
+          </FormGroup>
+          <FormGroup label="Port" fieldId="port">
+            <TextInput
+              value={formData?.port}
+              placeholder="Optional"
+              type="text"
+              id="source-port"
+              name="port"
+              onChange={event => handleInputChange('port', (event.target as HTMLInputElement).value)}
+              ouiaId="port"
+            />
+            <HelperText>Default port is 443</HelperText>
+          </FormGroup>
+        </React.Fragment>
+      )}
+      {isNetwork ? (
+        <FormGroup label="" fieldId="paramiko">
+          <Checkbox
+            key="paramiko"
+            label="Connect using Paramiko instead of Open SSH"
+            id="paramiko"
+            isChecked={formData?.useParamiko}
+            onChange={(_ev, checked) => handleInputChange('useParamiko', checked)}
+            ouiaId="options_paramiko"
+          />
+        </FormGroup>
+      ) : (
+        <React.Fragment>
+          <FormGroup label="Connection" fieldId="connection">
+            <SimpleDropdown
+              isFullWidth
+              label={formData?.sslProtocol || 'Select protocol'}
+              menuToggleOuiaId="options_ssl_protocol"
+              variant={'default'}
+              onSelect={item => handleInputChange('sslProtocol', item)}
+              dropdownItems={[
+                { item: 'SSLv23', ouiaId: 'sslv23' },
+                { item: 'TLSv1', ouiaId: 'tlsv1' },
+                { item: 'TLSv1.1', ouiaId: 'tlsv11' },
+                { item: 'TLSv1.2', ouiaId: 'tlsv12' },
+                { item: 'Disable SSL', ouiaId: 'disable_ssl' }
+              ]}
+            />
+          </FormGroup>
+          <FormGroup label="" fieldId="ssl_verify">
+            <Checkbox
+              key="ssl_verify"
+              label="Verify SSL certificate"
+              id="ssl_verify"
+              isDisabled={formData?.sslProtocol === 'Disable SSL'}
+              isChecked={formData?.sslProtocol !== 'Disable SSL' && formData?.sslVerify}
+              onChange={(_ev, checked) => handleInputChange('sslVerify', checked)}
+              ouiaId="options_ssl_cert"
+            />
+          </FormGroup>
+        </React.Fragment>
+      )}
+      <ActionGroup>
+        <Button variant="primary" onClick={onAdd}>
+          Save
+        </Button>
+        <Button variant="link" onClick={() => onClose()}>
+          Cancel
+        </Button>
+      </ActionGroup>
+    </Form>
+  );
+};
+
+const AddSourceModal: React.FC<AddSourceModalProps> = ({
+  isOpen,
+  source,
+  sourceType,
+  sourceName,
+  onClose = () => {},
+  onSubmit = () => {}
+}) => {
   return (
     <Modal
       variant={ModalVariant.small}
-      title={`${source ? 'Edit' : 'Add'} source: ${sourceType}`}
+      title={source ? `Edit Source: ${sourceName || ''}` : `Add Source: ${sourceType || ''}`}
```

Comment:
optional:
Declaring the name separate is definitely one path, it works! But we may be working harder than needed.

Since all we're doing when we call this component is using `sourceBeingEdited?.name` anyway it feels awkward to do that breakout before instead of doing it inside the component(s), like we do for editing. Doubly so since TS should be calling out any required props with our `SourceType`. 

If the concern is the optional chaining for `sourceBeingEdited?.name`, or the unit testing, I think we're still good since that application is more around the whole source being `undefined` rather than the attribute/prop being optional. And for the existing unit test it looks like that's already taken into account as a required prop in the mock.

```
title={source ? `Edit Source: ${source.name}` : `Add Source: ${sourceType || ''}`}
```

and if you don't like the old-skool format...

```
title={(source && `Edit Source: ${source.name}`) || `Add Source: ${sourceType || ''}`}
```

and if we want to adhere to graceful fallbacks for all responses =)

```
title={(source && `Edit Source: ${source.name || ''}`) || `Add Source: ${sourceType || ''}`}
```